### ISSUE TITLE
Slotted text node in scoped component is not relocated in the DOM properly when rendered with Maquette

### ISSUE THREAD

**Issue by eriklharper** [ionic-team/stencil#2801]
**Stencil version:**

```
 @stencil/core@2.4.0-3
```

**I'm submitting a:**

[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

**Current behavior:**

The second `calcite-label` element rendered using this code reproduces this issue, which is not placing the slotted text node in the proper place in the DOM:

```
<!DOCTYPE html>
<html dir="ltr" lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0" />
    <title>Stencil Component Starter</title>

    <script type="module" src="/build/stencil-starter-project-name.esm.js"></script>
    <script nomodule src="/build/stencil-starter-project-name.js"></script>
  </head>
  <body>
    <script type="module">
      import { h, createProjector } from "https://unpkg.com/maquette@3.3.7/dist/index.js";
      document.addEventListener('DOMContentLoaded', function () {
        const domNode = document.body;
        const projector = createProjector();
        const test = {
          renderMaquette: function () {
            return h("div", [
              h("calcite-label", ["I am a text node and won't get properly relocated"]),
              h("calcite-label", [h("span", ["I am a span and will get properly relocated"])]),
            ])
          }
        };
        projector.append(domNode, test.renderMaquette);
        projector.scheduleRender();
      });
    </script>
  </body>
</html>
```
![image](https://user-images.githubusercontent.com/821864/104657379-b69f7d00-5675-11eb-94ca-fde551780064.png)

**Expected behavior:**

Sending in plain text (without any wrapping element) to a scoped Stencil component properly places the content in the default `<slot />` position.

**Steps to reproduce:**

1. `git clone https://github.com/eriklharper/stencil-maquette-bug`
2. `npm i && npm start`
3. Open browser developer tools and inspect both rendered `calcite-label` elements.  Note that the first one is not placing the text content in the proper place while the second one is.

**Other information:**

Repro environment: https://github.com/eriklharper/stencil-maquette-bug

**Comment by julio8a** [comment]
@eriklharper, is there anything we can do on our end for this? or do we need to wait on Stencil?

**Comment by eriklharper** [comment]
> @eriklharper, is there anything we can do on our end for this? or do we need to wait on Stencil?

Yes, the workaround for this specific issue is to simply use a plain `<label>` in place of `<calcite-label>` anywhere that is having problems.  Currently this workaround is in place in the areas it is affecting based on what I've seen.  @AdelheidF can verify that as well on her end, but I have verified it several times and the workaround is solid for now while we wait for this fix.

**Comment by eriklharper** [comment]
@ionic-team have you had a chance to triage this?
